duplicate.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Using latent semantic indexing and similarity matrix
+
+import logging
+from collections import defaultdict
+
+import nltk
+from gensim import models, similarities
+from gensim.corpora import Dictionary
+from nltk.corpus import stopwords
+from nltk.stem.porter import PorterStemmer
+
+from bugbug import bugzilla
+
+logging.basicConfig(
+    format="%(asctime)s : %(levelname)s : %(message)s", level=logging.INFO
+)
+
+nltk.download("stopwords")
+
+# A map from bug id to its duplicate ids
+all_ids = defaultdict(int)
```

Comment:
You can use a set instead, since you just want to check for the existene of a given ID.

Reply:
isn't dictionary faster to check elements?

Reply:
A dictionary and a set are implemented almost in the same way, they are both really quick for checking the presence of an element. See https://wiki.python.org/moin/TimeComplexity.
The main difference is that you can store a value with the key in the case of a dict, and you can only store the key in the case of a set.